### PR TITLE
fix(angular/accordion): picking up lazy content from child component

### PR DESCRIPTION
**Pull request by mhaertwig**
We use a `ContentChild` to determine what lazy content to render inside an expansion panel. When the lazy content is nested further down inside another expansion panel, all ancestor panels up the tree will pick up the lowest lazy content from the lowest level. These changes add a check to ensure that the lazy content is rendered out by the closest panel.

See https://stackblitz.com/edit/angular-du1rkd?file=src%2Fapp%2Faccordion-nested-panel-lazy-content-example.html

### CODE REVIEW

File: src/angular/accordion/expansion-panel-base.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { CdkAccordionItem } from '@angular/cdk/accordion';
+import { InjectionToken } from '@angular/core';
+
+/**
+ * Base interface for a `MatExpansionPanel`.
```
Comment (jeripeierSBB) [inline src/angular/accordion/expansion-panel-base.ts:5]:
```suggestion
 * Base interface for a `SbbExpansionPanel`.
```

File: src/angular/accordion/expansion-panel-base.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * @docs-private
+ */
+export interface SBBExpansionPanelBase extends CdkAccordionItem {
```
Comment (jeripeierSBB) [inline src/angular/accordion/expansion-panel-base.ts:8]:
```suggestion
export interface SbbExpansionPanelBase extends CdkAccordionItem {
```

File: src/angular/accordion/expansion-panel-base.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  /** Whether the toggle indicator should be hidden. */
+  hideToggle: boolean;
+}
+
+/**
+ * Token used to provide a `SbbExpansionPanel` to `SbbExpansionPanelContent`.
+ * Used to avoid circular imports between `SbbExpansionPanel` and `SbbExpansionPanelContent`.
+ */
+export const SBB_EXPANSION_PANEL = new InjectionToken<SBBExpansionPanelBase>('SBB_EXPANSION_PANEL');
```
Comment (jeripeierSBB) [inline src/angular/accordion/expansion-panel-base.ts:17]:
```suggestion
export const SBB_EXPANSION_PANEL = new InjectionToken<SbbExpansionPanelBase>('SBB_EXPANSION_PANEL');
```

File: src/angular/accordion/expansion-panel-content.ts [hunk 4]
```diff
@@ -1,4 +1,8 @@
-import { Directive, TemplateRef } from '@angular/core';
+import { Directive, Inject, Optional, TemplateRef } from '@angular/core';
+import {
+  SBBExpansionPanelBase,
```
Comment (jeripeierSBB) [inline src/angular/accordion/expansion-panel-content.ts:3]:
```suggestion
  SbbExpansionPanelBase,
```

File: src/angular/accordion/expansion-panel-content.ts [hunk 5]
```diff
@@ -8,5 +12,8 @@ import { Directive, TemplateRef } from '@angular/core';
   selector: 'ng-template[sbbExpansionPanelContent]',
 })
 export class SbbExpansionPanelContent {
-  constructor(public _template: TemplateRef<any>) {}
+  constructor(
+    public _template: TemplateRef<any>,
+    @Inject(SBB_EXPANSION_PANEL) @Optional() public _expansionPanel?: SBBExpansionPanelBase
```
Comment (jeripeierSBB) [inline src/angular/accordion/expansion-panel-content.ts:17]:
```suggestion
    @Inject(SBB_EXPANSION_PANEL) @Optional() public _expansionPanel?: SbbExpansionPanelBase
```

File: src/angular/accordion/public-api.ts [hunk 6]
```diff
@@ -6,3 +6,4 @@ export { SbbExpansionPanelState, SbbExpansionPanel } from './expansion-panel';
 export * from './expansion-panel-header';
 export * from './expansion-panel-content';
 export * from './accordion-animations';
+export { SBB_EXPANSION_PANEL } from './expansion-panel-base';
```
Comment (jeripeierSBB) [inline src/angular/accordion/public-api.ts:9]: Would a star export also be possible?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

File: src/angular/accordion/public-api.ts (hunk 6 above)
Comment (mhaertwig) [reply src/angular/accordion/public-api.ts]: Yes, I've updated the PR.

**Review by jeripeierSBB — APPROVED** [review]
LGTM

**Review by kyubisation — APPROVED** [review]
LGTM 👍

**Pull request merged**